registration/tests/test_views.py

```diff
@@ -88,10 +86,15 @@ def _assert_registration_succeeds(self, form_data):
 
         # Test the email verification flow
         self._assert_email_verification_sent(application)
-        for verification_email in mail.outbox:  # fix flaky pylint: disable=no-member,useless-suppression
-            verify_url = re.search(r'https?://[^\s]+',
-                                   verification_email.body).group(0)
-            self.client.get(verify_url)
+        with patch('registration.provision._provision_instance', autospec=True) as mock_handler:
+            expected_call_count = 0
+            for verification_email in mail.outbox:  # fix flaky pylint: disable=no-member,useless-suppression
+                verify_url = re.search(r'https?://[^\s]+',
+                                       verification_email.body).group(0)
+                self.client.get(verify_url)
+                expected_call_count += 1
+            # Check to make sure we called _provision_instance when emails were verified.
+            self.assertEqual(mock_handler.call_count, expected_call_count)
```

Comment:
@haikuginger Argh, I just noticed this is testing the wrong thing. We don't really care if `_provision_instance` is called once or twice, or whatever. What we really want to know is (1) that the emails get marked as verified, and (2) that once *all* emails have been verified (but not before), does an instance get created and is [`spawn_appserver `](https://github.com/open-craft/opencraft/blob/f3f7e20/registration/provision.py#L67) called?

We might also want to test, say, that if an email address is re-verified multiple times, that no additional app servers get spawned.

(If these things tested elsewhere, then that's fine and if so, I'm not sure what value this change adds)